### PR TITLE
Fh 1722 appforms back end integration 

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
Appforms Back End Integration Feature Branch.

### CODE REVIEW

File: lib/utils/request.js [hunk 1]
```diff
@@ -22,6 +23,34 @@ var os = require("os");
...
+ */
+function createErrorMessage(statusCode, body){
+
+  var msg = 'Error - not 2xx status code. (' + statusCode + ')\n';
+
+  //If the body is a string, try to parse it as an object.
+  if(_.isString(body)){
+    try {
+      body = JSON.parse(body);
+    } catch(e){
```
Comment (fh-automatron) [inline lib/utils/request.js]: MAJOR Either remove or fill this block of code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEmptyBlock)

File: lib/cmd/fh3/services.js [hunk 2]
```diff
@@ -38,11 +39,36 @@ function services(argv, cb) {
   } else if ("read" === action) {
     if (args.length !== 2) return cb(services.usage);
     return readService(args[1], cb);
+  } else if ("data-sources" === action) {
+    if (args.length !== 2) return cb(services.usage);
```
Comment (fh-automatron) [inline lib/cmd/fh3/services.js]: MINOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-automatron** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by fh-automatron** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by fh-automatron** [comment]
@jasonmadigan @cianclarke please review this PR.

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* A component owner should review your change (we've tagged these above)
* This change should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* Ensure JIRA is up to date (e.g. deployment notes, tester notes, fix version, release notes)
* There should be an associated fhcap PR with this change - please provide a link to this
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

File: lib/cmd/fh3/services.js (hunk 2 above)
Comment (fh-qube) [inline lib/cmd/fh3/services.js]: MINOR Reformat the code to have only one statement per line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by lgriffin** [comment]
@nialldonnellyfh please be aware that https://github.com/feedhenry/fh-fhc/pull/195 will land before this and will need to be factored into your feature branch

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "request.js" has 516 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-fhc/blob/3e8332250753033e37a5c7dd2257c3cc6f769a6a/lib/utils/request.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "request.js" has 516 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-fhc/blob/1b6d961bde2a5dded999f78b6a7815670b48ecb4/lib/utils/request.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "request.js" has 516 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-fhc/blob/5b9b259fe3117cc188c71c1b10bce4790bc62a38/lib/utils/request.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Pull request merged**